### PR TITLE
feat: CDK Cross Region AWS CodeArtifact Sample

### PR DESCRIPTION
**Pull request by kareem-rady**
Added a Cross Region CodeArtifact sample utilizing CDK to deploy an upstream pipeline that packages and pushes a sample python package to another region hosted in a central Devops account and a downstream pipeline also hosted in a Devops account  that pulls the shared package and creates a lambda and a layer which is then deployed in an application account
This would be used to support a public APG with code hosted in this repo.

---

By submitting this pull request, I confirm that my contribution is made under the terms of the [Apache 2.0 license].

[Apache 2.0 license]: https://www.apache.org/licenses/LICENSE-2.0

### CODE REVIEW

File: python/cross-region-codeartifact/README.md [hunk 1]
```diff
@@ -0,0 +1,92 @@
+
+# CrossRegionArtifact Project
+
+## Introduction
+
+Customers utilize AWS CodeArtifact to store and distribute software packages , currently AWS CodeArtifact is only available in 13 regions listed [here](https://aws.amazon.com/codeartifact/faq/).
+
+In this guide we will will provide a demo on how to utilize AWS CodePipeline via AWS CodeBuild to build and publish an artifact to CodeArtifact and then consume the built artifact in another CodePipeline to deploy a lambda and layer built from the custom package hosted in a cross region CodeArtifact repository.
```
Comment (indrora) [inline python/cross-region-codeartifact/README.md:8]: Rewrite this in the form of "In this example, you will use AWS CodePipeline via AWS CodeBuild to build and publish and artifact..."

Include information on how customers can see the effects of the example as it does its thing. 

Include information on the usage of CDK Nags.

File: python/cross-region-codeartifact/README.md [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+
+Run:
+```
+cdk deploy --profile {aws_application_account_profile} crossaccount-role
+```
+    
+Followed by:
+```
+cdk deploy --profile {aws_devops_account_profile} downstream-pipeline
+```
```
Comment (indrora) [inline python/cross-region-codeartifact/README.md:69]: Explain what happens next. Explain how to interact with the deployed infrastructure.

File: python/cross-region-codeartifact/app.py [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+        """ Downstream Pipeline depends on crossaccount-role
+            and upstream-pipeline stacks being created first
+            Please run:
+            cdk deploy --profile {aws_profile} crossaccount-role
+            Followed by:
+            cdk deploy --profile {aws_profile} downstream-pipeline
+        """
+    )
+
+# adding cdk nag checks
```
Comment (indrora) [inline python/cross-region-codeartifact/app.py:69]: Expand this comment for someone who doesn't know.

File: python/cross-region-codeartifact/cdk/helpers/pipeline_common.py [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+        repository_name=repo_name,
+    )
+    return codecommit_repo
+
+
+def create_output_artifact():
+    return _codepipeline.Artifact()
+
+
+def update_role_with_codeartifact_policy(
```
Comment (indrora) [inline python/cross-region-codeartifact/cdk/helpers/pipeline_common.py:92]: Please rewrite this to use grant functions where possible. 

Please scope this as far down as possible where not using wildcards. 

We *highly* suggest against the use of raw IAM policies where role grants are more appropriate.

**Review by indrora — CHANGES_REQUESTED** [review]
Almost there. Needs some work with IAM policies.

**Comment by kareem-rady** [comment]
Changes have been added as requested

**Comment by indrora** [comment]
These changes don't really cover the comments and changes I've requested. In places where more information has been requested, little has changed. I asked for explanations as to why you aren't using grant_x calls (since you're working with L1 constructs, you don't have grant_x calls) and nothing exists in the README nor in the comments around it. 

Code quality wise, this doesn't pass a second or third read. You're generating ARNs by hand, which is a huge red flag in my opinion: You should be able to reference the resources directly across stacks without generating ARNs by hand; in cases with circular dependencies, references by name should be used where a direct reference isn't reasonable. Where an ARN is needed for a reference, use the appropriate call, such as [the ARN of a role](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_iam.Role.html#rolearn). 

There's very little documentation on what this code does. There's very little documentation on why this code does. The README does not clarify how the customer can use this and observe it working. 

I'm going to close this PR here. Feel free to clean this up and open a new PR.

**Pull request closed without merging**